chapter08/slc/ex2.pl

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Data::Dumper;
+use List::Util qw( reduce );
+
+print { $_->{filehandle} } @{ $_->{lines} } foreach values reduce { $a->{$b->[0]}{filehandle} //= IO::File->new( $b->[0] . '.info', 'w' ); push @{ $a->{$b->[0]}{lines} }, $b->[1]; $a; } {}, map { $_->[0] =~ s/\s+//g; [$_->[0], $_->[1]] } map { [lc((split /:/, $_)[0]), $_] } <>;
```

Comment:
Some weird indirection going on here, which fortunately for everyone else gets covered in a couple of chapters. Use of ``reduce`` an interesting one, that'll be useful to tell the group about.